hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeManager.java

```diff
@@ -511,7 +505,16 @@ private boolean isInactive(DatanodeInfo datanode) {
   private boolean isSlowNode(String dnUuid) {
     return avoidSlowDataNodesForRead && slowNodesUuidSet.contains(dnUuid);
   }
-  
+
+  public void setAvoidSlowDataNodesForReadEnabled(boolean enable) {
```

Comment:
We might need to check whether the slowPeerTracker is started or we might not get the slow peers.

Reply:
Consider slowNodesUuidSet is generated when the SlowPeerCollector thread is started，therefore  it is logical to judge Therefore, it is logical to judge whether the dnUuid exists in the slowNodesUuidSet?

Reply:
Thanks @haiyang1987 for your comment. I think the logic right now is no problem. 

I mean that when ```excludeSlowNodesEnabled``` is set to true we ```startSlowPeerCollector```, and ```stopSlowPeerCollector``` when ```excludeSlowNodesEnabled``` is set to false. There is no extra overhead. What do you think?

Reply:
Thanks @tomscut  for your comment. You have a point there.

But there is a problem，current parameter 'dataNodePeerStatsEnabled' and 'excludeSlowNodesEnabled' decision SlowPeerCollector thread whether to start ,But it didn't take into account avoid SlowDataNodesForRead logic.

And from the perspective of stability, the production environment of this function online can be consider divided into two stages：
The first is to start SlowPeerCollector thread, we can check whether the reported datanode is a slow node based on monitoring indicators.
Second, you can control whether to enable read/write avoid slow datanode according to dynamic parameters.

What do you think?

Reply:
> Thanks @tomscut for your comment. You have a point there.
> 
> But there is a problem，current parameter 'dataNodePeerStatsEnabled' and 'excludeSlowNodesEnabled' decision SlowPeerCollector thread whether to start ,But it didn't take into account avoid SlowDataNodesForRead logic.
> 
> And from the perspective of stability, the production environment of this function online can be consider divided into two stages： The first is to start SlowPeerCollector thread, we can check whether the reported datanode is a slow node based on monitoring indicators. Second, you can control whether to enable read/write avoid slow datanode according to dynamic parameters.
> 
> What do you think?

Thanks @haiyang1987 for your explanation. It is OK from my side. Can we also make ```excludeSlowNodesEnabled``` reconfigurable?

Reply:
Thanks @tomscut for your comment. 
HDFS-15120 already support BlockPlacementPolicyDefault#dfs.namenode.block-placement-policy.exclude-slow-nodes.enabled reconfigurable.  But the current way I think is heavy.  Do we need to support alone logic  make excludeSlowNodesEnabled reconfigurable？

Reply:
Thanks @haiyang1987 for your reminding, I didn't see the logic here before. I think we don't need to implement it separately if it's already supported.

Reply:
Thanks @tomscut for your comment.
Is there anything else that needs to be improved in the current code?

Reply:
> Thanks @tomscut for your comment. [HDFS-15120](https://issues.apache.org/jira/browse/HDFS-15120) already support BlockPlacementPolicyDefault#dfs.namenode.block-placement-policy.exclude-slow-nodes.enabled reconfigurable. But the current way I think is heavy. Do we need to support alone logic make excludeSlowNodesEnabled reconfigurable？

Hi @tomscut take a deep look at code discovery HDFS-15120, if we don't update 'dfs.block.replicator.classname', 'dfs.namenode.block-placement-policy.exclude-slow-nodes.enabled' can't support reconfigurable. 
so we need to support alone logic make excludeSlowNodesEnabled reconfigurable. 
Consider create a new issue to solve this problem. 
What do you think?